src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseExportService.java

```diff
@@ -228,28 +254,71 @@ public Path exportProgrammingExerciseRepositories(ProgrammingExercise exercise,
      * @return a zipped file
      */
     public Optional<File> exportInstructorRepositoryForExercise(long exerciseId, RepositoryType repositoryType, Path outputDir, List<String> exportErrors) {
+        var exerciseOrEmpty = loadExerciseForRepoExport(exerciseId, repositoryType.getName(), exportErrors);
+        if (exerciseOrEmpty.isEmpty()) {
+            return Optional.empty();
+        }
+
+        var exercise = exerciseOrEmpty.get();
+
+        String zippedRepoName = getZippedRepoName(exercise, repositoryType.getName());
+
+        var repositoryUrl = exercise.getRepositoryURL(repositoryType);
+
+        return exportRepository(repositoryUrl, repositoryType.getName(), zippedRepoName, exercise, outputDir, exportErrors);
+    }
+
+    /**
+     * Exports a repository available for an instructor/tutor for a given programming exercise. This can be a template,
+     * solution, or tests repository
+     *
+     * @param exerciseId     The id of the programming exercise that has the repository
+     * @param auxiliaryRepository the auxiliary repository to export
+     * @param outputDir The directory used for store the zip file
+     * @param exportErrors   List of failures that occurred during the export
+     * @return a zipped file
+     */
+    public Optional<File> exportInstructorAuxiliaryRepositoryForExercise(long exerciseId, AuxiliaryRepository auxiliaryRepository, Path outputDir, List<String> exportErrors) {
+        var exerciseOrEmpty = loadExerciseForRepoExport(exerciseId, auxiliaryRepository.getName(), exportErrors);
+        if (exerciseOrEmpty.isEmpty()) {
+            return Optional.empty();
+        }
+
+        var exercise = exerciseOrEmpty.get();
+
+        String zippedRepoName = getZippedRepoName(exercise, auxiliaryRepository.getRepositoryName());
+
+        var repositoryUrl = auxiliaryRepository.getVcsRepositoryUrl();
+
+        return exportRepository(repositoryUrl, auxiliaryRepository.getName(), zippedRepoName, exercise, outputDir, exportErrors);
+    }
```

Comment:
The similarity here makes me think that a common interface for both repository types might be beneficial. I don't know how often the code is so similar, though (it's up to the assignees to decide I guess).

Reply:
As far as I know, this is the only real similarity, I don't think a common interface is necessary for this case